timeless/views.py

```diff
@@ -68,48 +68,115 @@ def render_template(self, context):
+            model = models.ReservationSettings
+            template_name = "restaurants/tables/create_edit.html"
+            success_url_name = "reservation_settings_list"
+            not_found_url_name = "reservation_settings_list"
     """
-    def get_object(self):
-        """ Method for fetching object from db"""
-        raise NotImplementedError()
+    form = None
+    not_found_url_name = None
+    success_url_name = None
+
+    def get(self, id=None):
+        """
+        Get method fetch object from db and render it into template
+        """
+        assert self.form, "Form is required"
+        assert self.not_found_url_name, "not_found_url_name is required"
+
+        instance = self.get_object(id)
+        if not instance:
+            return redirect(url_for(self.not_found_url_name))
+
+        form = self.form.TableForm(request.form, instance=instance)
+        return self.render_template({"form": form})
+
+    def post(self, id=None):
+        """
+        Post method validate form from request and save it into db
+        """
+        form = self.form(request.form)
+        if form.validate():
+            form.save()
+            return redirect(url_for(self.success_url_name))
+        return self.render_template({"form": form})
+
+
+class CreateView(GenericView):
+    """Example
+        class SettingsCreateView(views.CreateView):
+            template_name = "restaurants/tables/create_edit.html"
+            success_url_name = "reservation_settings_list"
+            form = forms.SettingsForm
+    """
+    form = None
+    success_url_name = None
 
-    def get(self, *args, **kwargs):
-        """ Get method implementation """
-        raise NotImplementedError()
+    def get(self):
+        """Get method render create template """
+        assert self.form, "Form is required"
+        return self.render_template({"form": self.form()})
 
-    def post(self, *args, **kwargs):
-        """ Post method implementation"""
-        raise NotImplementedError()
+    def post(self):
+        """
+        Post method checks form validation,
+        save into db and redirect to success_url
+        """
+        assert self.form, "Form is required"
+        assert self.success_url_name, "Success_url_name is required"
+        form = self.form(request.form)
+        if form.validate():
+            form.save()
+        return redirect(url_for(self.success_url_name))
+
+
+class DeleteView(SingleObjectMixin, GenericView):
+    """
+    Example:
+    class SettingsDeleteView(views.DeleteView):
+        success_url_name = "reservation_settings_list"
+    """
+    success_url_name = None
+
+    def delete(self, id=None):
+        """
+        Calls the delete() method on the fetched object and then
+        redirects to the success URL.
+        """
+        instance = self.get_object(id)
+        instance.delete()
+        return redirect(url_for(self.success_url_name))
+
+    def post(self, id=None):
```

Comment:
@vladarefiev do we need a post method for deletion? Isn't the above ``delete`` method enough?

Reply:
@amihaiemil as I see you mean `delete` http method, in templates usually used only `get` and `post` methods.